src/shogun/features/CombinedDotFeatures.cpp

```diff
@@ -11,6 +11,8 @@
 
 using namespace shogun;
 
+const float64_t CCombinedDotFeatures::initial_weight = 1.0;
```

Comment:
Maybe just make it `constexpr`? Or does that only work in C++17?

Reply:
Yeah that would be best. It works for c++11 as well.

Reply:
ah ok, I think in C++17 what changes is that you can actually inline variables, so you could just have it the class declaration!

Reply:
you're right, it would need to be inlined, which would work only in C++17
https://dev.azure.com/shogunml/shogun/_build/results?buildId=1382

Reply:
but you can still replace the const with constexpr right?

Reply:
I tried just changing const to constexpr but it gave me compilation error
`constexpr static data member ‘initial_weight’ must have an initializer`
I also tried initializing it in the header file (which is ugly to my taste), it compiled locally but not with CI.

Reply:
I don't fully understand constexpr stuff, so I might be doing something wrong

Reply:
where did it fail in the CI? I just tried it in compiler explorer and it worked fine in C++14 (which it shouldn't).
constexpr has to be part of your declaration because it is evaluated at compile time, it's basically preprocessor with type safety. so you aren't "initialising" a variable, you are telling the compiler that is the value of the variable at runtime.

Reply:
makes sense. I messed up the times before, I didn't change `const` to `constexpr` in the declaration.

Reply:
so let's do it :)

Reply:
sorry, was waiting to write up the tests first